fennel/client_tests/test_dataset.py

```diff
@@ -1420,6 +1420,126 @@ def pipeline_join(cls, info: Dataset, sale: Dataset):
             ]
 
 
+@meta(owner="test@test.com")
+@source(webhook.endpoint("MovieRevenueWithRightFields"), disorder="14d", cdc="upsert")
+@dataset(index=True)
+class MovieRevenueWithRightFields:
+    movie: oneof(str, ["Jumanji", "Titanic", "RaOne", "ABC"]) = field(  # type: ignore
+        key=True
+    )
+    revenue: int
+    extra_field: int
+    t: datetime
+
+
+@meta(owner="satwant@fennel.ai")
+@dataset(index=True)
+class MovieStatsWithRightFields:
+    movie: oneof(str, ["Jumanji", "Titanic", "RaOne", "ABC"]) = field(  # type: ignore
+        key=True
+    )
+    rating: float
+    revenue_in_millions: float
+    t: datetime
+
+    @pipeline
+    @inputs(MovieRating, MovieRevenueWithRightFields)
+    def pipeline_join(cls, rating: Dataset, revenue: Dataset):
+        def to_millions(df: pd.DataFrame) -> pd.DataFrame:
+            df[str(cls.revenue_in_millions)] = df["revenue"] / 1000000
+            df[str(cls.revenue_in_millions)].fillna(-1, inplace=True)
+            return df[
+                [
+                    str(cls.movie),
+                    str(cls.t),
+                    str(cls.revenue_in_millions),
+                    str(cls.rating),
+                ]
+            ]
+
+        c = rating.join(revenue, how="left", on=[str(cls.movie)], right_fields=["revenue"])
```

Comment:
s/right_fields/rhs_fields

Reply:
I used right_fields because the param for join on is called right_on and not rhs_on.

Reply:
I wonder if we should just use `fields` here since that's quite disambiguated on its own (there is no analog of left_fields)? cc @aditya-nambiar